### PR TITLE
DNS searchdomains pod to DHCP

### PR DESCRIPTION
**Pull request by mlsorensen**
Pass pod's DNS search domains up to VM via DHCP server. If none are found, it defaults to 'cluster.local'. This addresses #841

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/network/network.go [hunk 1]
```diff
@@ -167,19 +171,34 @@ func (h *NetworkUtilsHandler) StartDHCP(nic *VIF, serverAddr *netlink.Addr) {
...
+		return nil, nil, err
+	}
+
+	searchDomains, err := ParseSearchDomains(string(b))
+	if err != nil {
+		return nil, nil, err
 	}
 
-	return ParseNameservers(string(b))
+	log.Log.Reason(err).Errorf("Found nameservers in %s: %s", resolvConf, bytes.Join(nameservers, []byte{' '}))
```
Comment (davidvossel) [inline pkg/virt-launcher/virtwrap/network/network.go]: Infof instead of Errorf here?

File: pkg/virt-launcher/virtwrap/network/network.go [hunk 2]
```diff
@@ -214,6 +233,32 @@ func ParseNameservers(content string) ([][]byte, error) {
...
 
+func ParseSearchDomains(content string) ([]string, error) {
+	var searchDomains []string
+
+	scanner := bufio.NewScanner(strings.NewReader(content))
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, domainSearchPrefix) {
+			doms := strings.Fields(strings.TrimLeft(line, domainSearchPrefix))
```
Comment (davidvossel) [inline pkg/virt-launcher/virtwrap/network/network.go]: use strings.TrimPrefix instead. TrimLeft isn't doing what you think it's doing here. 

https://stackoverflow.com/questions/29187086/why-trimleft-doesnt-work-as-expected

File: pkg/virt-launcher/virtwrap/network/dhcp/dhcp.go [hunk 3]
```diff
@@ -31,7 +34,15 @@ import (
...
-const infiniteLease = 999 * 24 * time.Hour
+const (
+	infiniteLease             = 999 * 24 * time.Hour
+	errorSearchDomainNotValid = "Search domain is not valid"
+	errorSearchDomainTooLong  = "Search domains length exceeded allowable size"
+)
+
+// simple domain validation regex. Put it here to avoid compiling each time.
+// Note this requires that unicode domains be presented in their ASCII format
+var searchDomainValidationRegex = regexp.MustCompile(`^(?:[_a-z0-9](?:[_a-z0-9-]{0,61}[a-z0-9])?\.)+(?:[a-z](?:[a-z0-9-]{0,61}[a-z0-9])?)?$`)
```
Comment (stu-gott) [inline pkg/virt-launcher/virtwrap/network/dhcp/dhcp.go]: If I'm reading this regex correctly, It appears that it's enforcing that labels cannot end with a hyphen or underscore. That's definitely correct for hyphens, but I am unsure of underscores. (RFC 1123 specifically supersedes RFC 952 with regards to [digits in] the first character, but as near as I can tell is ambiguous with regards to subsequent ones).

**Comment by davidvossel** [comment]
ok to test

**Comment by davidvossel** [comment]
retest this please

**Pull request merged**